README.md

```diff
@@ -145,11 +145,13 @@ type Message = StartMessage | DefineMessage | EvaluateMessage | AnalysisMessage;
 
 interface DefineResponse extends Base {
   success: boolean;
+  error?: string;
 }
 
 interface EvaluateResponse extends Base {
   output: any;
   timings?: Timing[];
+  error?: string;
```

Comment:
Same request here; I think also, `output` should be made optional, and a mandatory `success` field should be added.

Reply:
Fiddling with `output` belongs in a different PR.

Reply:
Hm? I'm not sure I understand; are you saying that in this PR, even if a tool returns an `error`, it must _also_ return an `output`?

Reply:
That does seem weird, yes. At the Python level it is a `None` value, but I don't actually know how that is serialised to JSON.

If we make it optional, what do we do about an `EvaluateResponse` that has neither `output` nor `error`?

Reply:
Neither `output` nor `error` is fine, as long as `"success"` is `false`. If `"success": true` but no `output` then that's a protocol error.

Reply:
Alright, I'll see if I can do that. This really would be better if Python could express sum types.

Reply:
I think Python can express sum types more or else fine; this is really just a matter of whether we want to include that sum type in the protocol or not. I think it could be reasonable to do so.